Source/engine/random.cpp

```diff
@@ -13,6 +16,40 @@ uint32_t sglGameSeed;
 /** Borland C/C++ psuedo-random number generator needed for vanilla compatibility */
 std::linear_congruential_engine<uint32_t, 0x015A4E35, 1, 0> diabloGenerator;
 
+/** Xoshiro pseudo-random number generator to provide less predictable seeds */
+xoshiro128plusplus seedGenerator;
+
+uint32_t xoshiro128plusplus::rotl(const uint32_t x, int k)
+{
+	return std::rotl(x, k);
+}
+
+uint64_t xoshiro128plusplus::timeSeed()
+{
+	auto now = std::chrono::system_clock::now();
+	auto nano = std::chrono::nanoseconds(now.time_since_epoch());
+	long long time = nano.count();
+	SplitMix64 seedSequence { static_cast<uint64_t>(time) };
+	return seedSequence.next();

```

Comment:
this is half of `seed(uint64_t)` already, could just call `this->seed(time)` instead?